packages/flutter/test/material/bottom_navigation_bar_test.dart

```diff
@@ -100,19 +100,99 @@ void main() {
 
     const double selectedFontSize = 14.0;
     const double unselectedFontSize = 12.0;
-    expect(tester.renderObject<RenderParagraph>(find.text('AC')).text.style.fontSize, selectedFontSize);
+    final TextStyle selectedFontStyle = tester.renderObject<RenderParagraph>(find.text('AC')).text.style;
+    final TextStyle unselectedFontStyle = tester.renderObject<RenderParagraph>(find.text('Alarm')).text.style;
+    expect(selectedFontStyle.color, equals(primaryColor));
+    expect(selectedFontStyle.fontSize, selectedFontSize);
+    expect(selectedFontStyle.fontWeight, isNull);
+    expect(selectedFontStyle.height, isNull);
+    expect(unselectedFontStyle.color, equals(captionColor));
+    expect(unselectedFontStyle.fontWeight, isNull);
+    expect(unselectedFontStyle.height, isNull);
     // Unselected label has a font size of 14 but is scaled down to be font size 12.
-    expect(tester.renderObject<RenderParagraph>(find.text('Alarm')).text.style.fontSize, selectedFontSize);
     expect(
       tester.firstWidget<Transform>(find.ancestor(of: find.text('Alarm'), matching: find.byType(Transform))).transform,
       equals(Matrix4.diagonal3(Vector3.all(unselectedFontSize / selectedFontSize))),
     );
-    expect(tester.renderObject<RenderParagraph>(find.text('AC')).text.style.color, equals(primaryColor));
-    expect(tester.renderObject<RenderParagraph>(find.text('Alarm')).text.style.color, equals(captionColor));
     expect(_getOpacity(tester, 'Alarm'), equals(1.0));
     expect(_getMaterial(tester).elevation, equals(8.0));
   });
 
+  testWidgets('Custom selected and unselected font styles', (WidgetTester tester) async {
+    const TextStyle selectedTextStyle = TextStyle(fontWeight: FontWeight.w200, fontSize: 18.0);
```

Comment:
I was told you can drop the `.0` now if the analyzer can figure out the type implicitly.

Reply:
Dart supports this feature, but as far as I have seen, the flutter framework code has not switched over to this pattern yet. I'd imagine it will be done as a bulk change similar to when the `new` keyword became optional.